### PR TITLE
add default project domain in execute launch plan

### PR DESCRIPTION
**Pull request by troychiu**
# TL;DR
1. add default project domain in execute launch plan
2. add name as a parameter
## Type
 - [X] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
When I executed a launch plan with FlyteRemote, it didn't run with default project domain of the remote but needed specified project domain. So, I fixed it with if there is no specified project domain, use default project domain of the remote.

## Tracking Issue
https://github.com/flyteorg/flyte/issues/4249

## Follow-up issue
_NA_

### CODE REVIEW

**Review by wild-endeavor — COMMENTED** [review]
should we just add this to the top level def execute instead?

**Comment by troychiu** [comment]
> should we just add this to the top level def execute instead?

Hi @wild-endeavor, do you mean we can resolve the identifiers before actually classify different types of entities?
For example, move the `resolved_identifiers = self._resolve_identifier_kwargs(entity, project, domain, name, version)`
to https://github.com/flyteorg/flytekit/blob/74d63d7265383c2af07952e835bcebf95b2d0942/flytekit/remote/remote.py#L1158

**Review by pingsutw — APPROVED** [review]
